homeassistant/components/webostv/media_player.py

```diff
@@ -202,17 +345,40 @@ def _update_states(self) -> None:
         self._attr_source = self._current_source
         self._attr_source_list = sorted(self._source_list)
 
+        if (
+            (self.language_code == "")
+            and (self._client.software_info is not None)
+            and (self._client.software_info.get("language_code") is not None)
+            and (len(self._client.software_info.get("language_code")) > 4)
```

Comment:
Do we think we need to validate this? And why was `> 4` chosen? Just want to see if we're appropriately following the spec for these language codes/tags

Reply:
How can a developer be sure what a 3rd application/module set as output?
I'm not sure "language_code" is the same for all implementations and WebOS version, so I usually set constraints like that especially when INSIDE it there could be a big comparison for all other Codes. If in some implementation "language_code" returns "ITA" or "ENG" there isn't any reason to do match comparison against "it-IT", "us-US " and so on, so this stops any further check before the string comparison.

I think it's common in code-development setting constraints more stricted than could be required especially if source is not under our control ;)

Reply:
Thats fair, my question was more around if we're going to validate the returned string, why just validate the length and not be stricter? Such as a regex match for `^[a-z]{2}-[A-Z]{2}$`